fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -154,61 +169,98 @@ public FHIRPersistenceJDBCImpl() throws Exception {
         }
         this.updateCreateEnabled = fhirConfig.getBooleanProperty(PROPERTY_UPDATE_CREATE_ENABLED, Boolean.TRUE);
         this.userTransaction = retrieveUserTransaction(TXN_JNDI_NAME);
+        
+        if (userTransaction != null) {
+            this.trxSynchRegistry = getTrxSynchRegistry();
+        } else {
+            this.trxSynchRegistry = null;
+        }
 
         ParameterNamesCache.setEnabled(fhirConfig.getBooleanProperty(PROPERTY_JDBC_ENABLE_PARAMETER_NAMES_CACHE,
                                        Boolean.TRUE));
         CodeSystemsCache.setEnabled(fhirConfig.getBooleanProperty(PROPERTY_JDBC_ENABLE_CODE_SYSTEMS_CACHE,
                                     Boolean.TRUE));
         ResourceTypesCache.setEnabled(fhirConfig.getBooleanProperty(PROPERTY_JDBC_ENABLE_RESOURCE_TYPES_CACHE,
                                       Boolean.TRUE));
-
-        sharedConnection = this.createConnection();
-        this.resourceDao = FHIRResourceDAOFactory.getResourceDAO(sharedConnection, this.getTrxSynchRegistry());
-        this.parameterDao = new ParameterDAOImpl(this.getTrxSynchRegistry());
+        
+        //this.connectionStrategy = new FHIRDbTenantDatasourceConnectionStrategy(getTrxSynchRegistry(), buildActionChain());
```

Comment:
Do we need this ?

Reply:
I've removed for now, because we'll need something a little more sophisticated when we implement the simple datasource feature.